zeth_contracts/contracts/BLS12_377MixerBase.sol

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2015-2020 Clearmatics Technologies Ltd
+//
+// SPDX-License-Identifier: LGPL-3.0+
+
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "./MixerBase.sol";
+import "./MiMC31.sol";
+
+// Partial implementation of abstract MixerBase which implements the
+// curve-specific methods to use the BLS12-377 pairing.
+contract BLS12_377MixerBase is MixerBase
+{
+    // TODO: Code here is very similar to AltBN128MixerBase, with only the
+    // constants changed. Look into sharing more code (possibly by making some
+    // of these constants dynamic).
+
+    // Constants regarding the hash digest length, the prime number used and
+    // its associated length in bits and the max values (v_in and v_out)
+    // field_capacity = floor( log_2(r) )
+    uint256 constant field_capacity = 252;
+
+    // Number of residual bits per bytes32
+    uint256 constant num_residual_bits = digest_length - field_capacity;
+
+    // Shift to move residual bits from lowest order to highest order
+    uint256 constant residual_bits_shift = 256 - num_residual_bits;
+
+    // Mask to extract the residual bits in the high-order position
+    uint256 constant residual_bits_mask =
+    ((1 << num_residual_bits) - 1) << residual_bits_shift;
+
+    // Total number of residual bits from packing of 256-bit long string into
+    // 253-bit long field elements to which are added the public value of size
+    // 64 bits
+    uint256 constant total_num_residual_bits =
```

Comment:
That's a side comment that will need to be handled in the broader context of our contracts (and code base in general). May be worth sticking closer to the specs terminology. This constant is denoted: `RSDBLEN` in the specs:
1. It's ALLCAPS which is consistent with the solidity coding standards for constants
2. Renaming according to the specs - when possible and meaningful - shortens the bridge between "specs and implementation" which can only be a good thing

Reply:
RENAMED and made all constants UPPER_SNAKE_CASE as in the new conventions.  I didn't use `RSDBLEN` itself, since it's not really in the style of the other constants and could look pretty opaque, but referenced it in the comments.

Reply:
> RENAMED and made all constants UPPER_SNAKE_CASE as in the new conventions

Perfect, thanks for that.

> I didn't use RSDBLEN itself, since it's not really in the style of the other constants and could look pretty opaque, but referenced it in the comments.

A comment like this to explicitly refer to a specific part of the specs makes the job perfectly.